core/controllers/reader.py

```diff
@@ -324,24 +381,32 @@ def post(self, exploration_id):
                 issue_customization_args = issue['issue_customization_args']
                 # In case issue_keyname is 'state_names', the ordering of the
                 # list is important i.e. [a,b,c] is different from [b,c,a].
-                issue_keyname = feconf.ISSUE_TYPE_KEYNAME_MAPPING[issue[
+                issue_keyname = stats_models.ISSUE_TYPE_KEYNAME_MAPPING[issue[
                     'issue_type']]
                 if (issue_customization_args[issue_keyname] ==
                         customization_args[issue_keyname]):
+                    issue_found = True
                     if (len(issue['playthrough_ids']) <
                             feconf.MAX_PLAYTHROUGHS_FOR_ISSUE):
-                        playthrough_id = (
-                            stats_models.PlaythroughModel.create_from_dict(
-                                playthrough_data))
+                        playthrough_id = stats_models.PlaythroughModel.create(
+                            playthrough_data['exp_id'],
+                            playthrough_data['exp_version'],
+                            playthrough_data['issue_type'],
+                            playthrough_data['issue_customization_args'],
+                            playthrough_data['playthrough_actions'],
+                            playthrough_data['is_valid'])
                         exp_issues.unresolved_issues[index][
                             'playthrough_ids'].append(playthrough_id)
-                    issue_found = True
                     break
 
         if not issue_found:
-            playthrough_id = (
-                stats_models.PlaythroughModel.create_from_dict(
-                    playthrough_data))
+            playthrough_id = stats_models.PlaythroughModel.create(
+                playthrough_data['exp_id'],
+                playthrough_data['exp_version'],
+                playthrough_data['issue_type'],
+                playthrough_data['issue_customization_args'],
+                playthrough_data['playthrough_actions'],
+                playthrough_data['is_valid'])
             issue = {
```

Comment:
This should be an Issue domain object, right? Not a dict? (Otherwise unresolved_issues should be unresolved_issues_dicts.)

Generally we convert to dicts only when storing as jsonproperty, but when retrieving then we convert everything (+ subdicts) into domain objects and work with domain objects throughout the logic layer and higher.

Reply:
Done. Required a major revamp to change all sub dicts into sub domain objects, so PTAL @seanlip